scripts/hooks/pre-commit.sh

```diff
@@ -30,5 +30,24 @@ function lint_and_add_modified_go_files() {
   done
 }
 
+function is_proto_all_required() {
+  local before_files=$(git status --porcelain | awk '{print $2}')
+  make proto-all
+  local after_files=$(git status --porcelain | awk '{print $2}')
+  local changed_files=$(comm -13 <(echo "$before_files" | sort) <(echo "$after_files" | sort))
+  if [[ -n "$changed_files" ]]; then
+    echo "Error: Please run 'make proto-all' and commit the updated files."
+    
+    # Revert the changes made by make proto-all
+    for file in $changed_files; do
```

Comment:
A drawback I can see with this approach is that the pre commit hook becomes quite slower.
I was thinking of the approach to just check if any `.proto` file is changed but this doesn't guarantee whether the command should be run

Reply:
Refactored. Now the script checks in the unstaged .proto files. If it finds a change it runs `make proto-all` and logs the files that were changed. If it doesn't find a change in protos then it just exits